src/test/java/com/znsio/sample/e2e/businessLayer/amazon/HomePageBL.java

```diff
@@ -0,0 +1,48 @@
+package com.znsio.sample.e2e.businessLayer.amazon;
+
+import com.context.TestExecutionContext;
+import com.znsio.e2e.entities.Platform;
+import com.znsio.e2e.runner.Runner;
+import com.znsio.sample.e2e.entities.SAMPLE_TEST_CONTEXT;
+import com.znsio.sample.e2e.screen.amazon.HomePageScreen;
+import org.apache.log4j.Logger;
+import org.assertj.core.api.SoftAssertions;
+
+public class HomePageBL {
+
+    private static final Logger LOGGER = Logger.getLogger(HomePageBL.class.getName());
+    private final TestExecutionContext context;
+    private final SoftAssertions softly;
+    private final String currentUserPersona;
+    private final Platform currentPlatform;
+
+    public HomePageBL(String userPersona, Platform forPlatform){
+        long threadId = Thread.currentThread()
+                .getId();
+        this.context = Runner.getTestExecutionContext(threadId);
+        softly = Runner.getSoftAssertion(threadId);
+        this.currentUserPersona = userPersona;
+        this.currentPlatform = forPlatform;
+        Runner.setCurrentDriverForUser(userPersona, forPlatform, context);
+    }
+
+    public HomePageBL() {
+        long threadId = Thread.currentThread()
+                .getId();
+        this.context = Runner.getTestExecutionContext(threadId);
+        softly = Runner.getSoftAssertion(threadId);
+        this.currentUserPersona = SAMPLE_TEST_CONTEXT.ME;
+        this.currentPlatform = Runner.platform;
+    }
+
+    public HomePageBL launchHomePageAmazon() {
+        boolean isAmazonHomePageLaunched1 = HomePageScreen.get().isAmazonHomePageLaunched();
+        return this;
+    }
+
+    public SearchBL userSearchesForProduct(String productName) {
+        HomePageScreen.get().searchForProduct(productName);
+        return new SearchBL();
```

Comment:
assertion is missing

Reply:
Should I change the return type of method "HomePageScreen.get().searchForProduct(productName)" to boolean to add assertion here or should I add an assertion for user's navigation to search list page using

 assertThat(SearchResultScreen.get().isUserOnSearchResultPage()).as("User is on search result page").isTrue();

Please suggest.